torch/csrc/Exceptions.h

```diff
@@ -360,33 +367,52 @@ struct PyWarningHandler {
     in_exception_ = true;
   }
 
+  // Trigger processing of warnings
+  TORCH_PYTHON_API void process_warnings();
+
  private:
   InternalHandler internal_handler_;
   at::WarningHandler* prev_handler_;
   bool in_exception_;
 };
 
 namespace detail {
+
+struct noop_gil_scoped_release {
+  noop_gil_scoped_release() {
+    // suppress `unused variable` error messages at call sites
+    (void)(this != (this + 1));
+  }
+};
+
+template <bool release_gil>
+using conditional_gil_scoped_release = std::conditional_t<
```

Comment:
Is this just a refactor to simplify the code below?

Reply:
Yes. It makes it a compile-time check instead of run-time check and avoids an extra condition in the code below to avoid excess C&P